### PR TITLE
Introduce AssetProvider / AssetService

### PR DESCRIPTION
**Pull request by wbreza**
-Introduced AssetProvider (subset of storage provider)
-Added baseline Bing Search Image asset provider
-Added baseline editor page to load from underlying asset provider
-Unit tests.

### CODE REVIEW

File: src/App.test.tsx [hunk 1]
```diff
@@ -4,7 +4,7 @@ import App from "./App";
 import { Provider } from "react-redux";
 import createReduxStore from "./store/store";
 import initialState from "./store/initialState";
-import ApplicationState from "./store/applicationState";
+import { IApplicationState as ApplicationState } from "./store/applicationState";
```
Comment (tbarlow12) [inline src/App.test.tsx]: Do we care about being consistent with the aliases? Some interfaces are renamed while others are not. Any specific reason?

File: src/components/pages/connections/connectionForm.tsx [hunk 2]
```diff
@@ -69,16 +69,18 @@ export default class ConnectionForm extends React.Component<IConnectionFormProps
 
     private bindForm(connection: IConnection, resetProviderOptions: boolean = false) {
         const providerType = connection ? connection.providerType : null;
+        let newFormSchema: any = this.state.formData;
```
Comment (tbarlow12) [inline src/components/pages/connections/connectionForm.tsx]: Did you mean `this.state.formSchema` here?

File: src/components/pages/editorPage/editorPage.tsx [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+    public render() {
+        const { project } = this.props;
+        const { assets, selectedAsset } = this.state;
+
+        if (!project) {
+            return (<div>Loading...</div>);
+        }
+
+        return (
+            <div className="editor-page">
```
Comment (tbarlow12) [inline src/components/pages/editorPage/editorPage.tsx]: Do we want to break this down into sub-components? I'm fine to keep it like this for now, but sub-components might make it more extensible in the future

File: src/App.test.tsx (hunk 1 above)
Comment (wbreza) [reply src/App.test.tsx]: fixed.

File: src/components/pages/connections/connectionForm.tsx (hunk 2 above)
Comment (wbreza) [reply src/components/pages/connections/connectionForm.tsx]: Good catch! - fixed.

File: src/components/pages/editorPage/editorPage.tsx (hunk 3 above)
Comment (wbreza) [reply src/components/pages/editorPage/editorPage.tsx]: Leaving that to the team - didn't want to step on other ongoing work in this area.

**Pull request merged**